clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -760,18 +760,43 @@ public FetchManagerMetrics(Metrics metrics, String metricGrpPrefix) {
         }
 
         public void recordTopicFetchMetrics(String topic, int bytes, int records) {
+            Map<String, String> metricTags = new HashMap<>();
+            metricTags.put("topic", "topic-" + topic.replace(".", "_"));
```

Comment:
Either way works for me. Now I'm thinking about why the Selector metrics had the "node-" prefix. Maybe it's only because the node ids were simple integers before.